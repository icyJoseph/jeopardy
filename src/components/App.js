import React, { Component } from "react";
import { Button } from "reactstrap";
import * as constants from "../constants";

class App extends Component {
  render() {
    return (
      <div>
        <h2 className="display-1">{constants.mainTitle}</h2>
        <Button className="btn-get-categories">{constants.GET_CATS_BTN}</Button>
      </div>
    );
  }
}

export default App;
