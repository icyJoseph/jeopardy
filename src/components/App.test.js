import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("App", () => {
  let app = shallow(<App />);

  it("renders properly", () => {
    expect(app).toMatchSnapshot();
  });

  it("has a display-1 heading", () => {
    expect(app.find(".display-1").exists()).toBe(true);
  });
});
