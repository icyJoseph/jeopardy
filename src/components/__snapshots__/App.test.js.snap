// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="display-1"
>
          Welcome to Jeopardy
</h2>,
        <Button
          className="btn-get-categories"
          color="secondary"
          tag="button"
>
          Get Jeopardy Categories
</Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to Jeopardy",
          "className": "display-1",
        },
        "ref": null,
        "rendered": "Welcome to Jeopardy",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Get Jeopardy Categories",
          "className": "btn-get-categories",
          "color": "secondary",
          "tag": "button",
        },
        "ref": null,
        "rendered": "Get Jeopardy Categories",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="display-1"
>
            Welcome to Jeopardy
</h2>,
          <Button
            className="btn-get-categories"
            color="secondary"
            tag="button"
>
            Get Jeopardy Categories
</Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Jeopardy",
            "className": "display-1",
          },
          "ref": null,
          "rendered": "Welcome to Jeopardy",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Get Jeopardy Categories",
            "className": "btn-get-categories",
            "color": "secondary",
            "tag": "button",
          },
          "ref": null,
          "rendered": "Get Jeopardy Categories",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
